MQTTnet.Core/Client/MqttPacketDispatcher.cs

```diff
@@ -10,7 +10,7 @@ namespace MQTTnet.Core.Client
     public class MqttPacketDispatcher
     {
         private readonly object _syncRoot = new object();
-        private readonly List<MqttBasePacket> _receivedPackets = new List<MqttBasePacket>();
+        private readonly HashSet<MqttBasePacket> _receivedPackets = new HashSet<MqttBasePacket>();
```

Comment:
there is no need for hashcodes the packets are same references and so have same hashcode

Reply:
just start the perf test app it will take some time and then throughput drops to zero. if that is the case hit pause and look at the msgdispatcher in my case there where thousands of objcts in the list and swichting to hashset resolved the issue

Reply:
and here is the line that adds the same package to the list

https://github.com/chkr1011/MQTTnet/blob/16949ffb218291fbc07cd8a7800e2aa2085c0e76/MQTTnet.Core/Client/MqttPacketDispatcher.cs#L54

Reply:
alternativily the list must be cleared here:

https://github.com/chkr1011/MQTTnet/blob/96b92aabae156beb99885b0a644da3e16a7d5178/MQTTnet.Core/Client/MqttPacketDispatcher.cs#L103

Reply:
![image](https://user-images.githubusercontent.com/6747376/30265250-e74fc0f2-96db-11e7-9321-bf6e99b1806e.png)

Reply:
@chkr1011 ping

Reply:
Yes you are right. Now I see it :-)